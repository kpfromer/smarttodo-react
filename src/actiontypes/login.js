export const INVALIDATE_LOGIN = 'login/INVALIDATE_LOGIN';
export const REQUEST_LOGIN = 'login/REQUEST_LOGIN';
export const RECEIVE_LOGIN = 'login/RECEIVE_LOGIN';
export const FAILURE_LOGIN = 'login/FAILURE_LOGIN';