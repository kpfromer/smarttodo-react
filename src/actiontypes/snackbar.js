export const CLOSE_SNACKBAR = 'snackbar/CLOSE_SNACKBAR';
export const CHANGE_SNACKBAR = 'snackbar/CHANGE_SNACKBAR';