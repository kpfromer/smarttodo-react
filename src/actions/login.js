import * as LoginActionTypes from '../actiontypes/login';
import { CALL_API } from "../middleware/api/api";

export const tryLogin = (username, password) => ({
  [CALL_API]: {
    types: [
      LoginActionTypes.REQUEST_LOGIN,
      LoginActionTypes.RECEIVE_LOGIN,
      LoginActionTypes.FAILURE_LOGIN
    ],
    method: 'POST',
    endpoint: '/login',
    data: { username, password },
    authenticate: false,
    mapResponse: response => response.access_token
  }
});

export const logout = () => ({
  type: LoginActionTypes.INVALIDATE_LOGIN
});