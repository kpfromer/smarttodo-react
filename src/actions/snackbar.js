import { CHANGE_SNACKBAR, CLOSE_SNACKBAR } from '../actiontypes/snackbar';

export const closeSnackbar = () => ({
  type: CLOSE_SNACKBAR
});

export const changeSnackbar = ({ message, variant, timing = 3000 }) => ({
  type: CHANGE_SNACKBAR,
  message,
  variant,
  timing
});