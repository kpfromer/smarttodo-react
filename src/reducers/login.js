import * as LoginActionTypes from '../actiontypes/login';

function login
(
  state = {
    isFetching: false,
    didInvalidate: false,
    token: localStorage.getItem('id') || null
  },
  action
) {
  switch (action.type) {
    case LoginActionTypes.INVALIDATE_LOGIN:
      localStorage.removeItem('id');
      return {
        ...state,
        token: null,
        didInvalidate: true
      };
    case LoginActionTypes.REQUEST_LOGIN:
      return {
        ...state,
        isFetching: true,
        didInvalidate: false
      };
    case LoginActionTypes.RECEIVE_LOGIN:
      localStorage.setItem('id', action.response);
      return {
        ...state,
        isFetching: false,
        didInvalidate: false,
        token: action.response
      };
    case LoginActionTypes.FAILURE_LOGIN:
      return {
        ...state,
        isFetching: false
      };
    default:
      return state;
  }
}

export default login