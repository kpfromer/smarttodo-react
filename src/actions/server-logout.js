import { logout } from "./login";
import { changeSnackbar } from "./snackbar";

export const serverLogout = dispatch => {
  dispatch(changeSnackbar({
    message: 'Redirecting to login. You were logged out.',
    variant: 'info'
  }));
  dispatch(logout());
};