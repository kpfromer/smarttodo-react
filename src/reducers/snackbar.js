import { CHANGE_SNACKBAR, CLOSE_SNACKBAR } from '../actiontypes/snackbar';

export function snackbar
(
  state = {
    isActive: false,
    message: '',
    variant: '',
    timing: 5000
  },
  action
) {
  switch (action.type) {
    case CLOSE_SNACKBAR:
      return {
        ...state,
        isActive: false
      };
    case CHANGE_SNACKBAR:
      return {
        ...state,
        isActive: true,
        message: action.message,
        variant: action.variant,
        timing: action.timing
      };
    default:
      return state;
  }
}

export default snackbar;