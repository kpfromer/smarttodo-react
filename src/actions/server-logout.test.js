import * as loginActions from './login';
import * as snackbar from "./snackbar";
import { serverLogout } from "./server-logout";

jest.mock('./snackbar', () => ({
  changeSnackbar: jest.fn()
}));

jest.mock('./login', () => ({
  logout: jest.fn()
}));

describe('serverLogout', () => {
  let action, mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.fn();
    snackbar.changeSnackbar.mockReturnValue('changeSnackbarAction');
    loginActions.logout.mockReturnValue('logoutAction');
    action = serverLogout(mockDispatch);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });


  it('changes snackbar message', () => {
    expect(snackbar.changeSnackbar.mock.calls[0][0]).toMatchSnapshot();
    expect(mockDispatch.mock.calls[0][0]).toBe('changeSnackbarAction');
  });
  it('then logouts user', () => {
    expect(mockDispatch.mock.calls[1][0]).toBe('logoutAction');
  });
});