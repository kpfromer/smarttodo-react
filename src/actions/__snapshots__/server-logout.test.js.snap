// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serverLogout changes snackbar message 1`] = `
Object {
  "message": "Redirecting to login. You were logged out.",
  "variant": "info",
}
`;
